BetaCameras/Sick.VisionaryT.Pro/VisionaryTPro.cs

```diff
@@ -313,22 +300,19 @@ private ProjectiveTransformationZhang ParseIntrinsics(CameraObject metaObj)
                 );
         }
 
-        private unsafe FloatCameraImage ParseImage(string base64Data, float scaleFactor = 1000.0f)
+        private unsafe FloatCameraImage ParseImage(char[] base64Data, int offset, int length, float scaleFactor = 1000.0f)
         {
-            byte[] raw = Convert.FromBase64String(base64Data);
+            byte[] raw = Convert.FromBase64CharArray(base64Data, offset, length);
 
             FloatCameraImage image = new FloatCameraImage(_imageWidth, _imageHeight);
 
             fixed (byte* rawData = raw)
             {
                 ushort* ushortData = (ushort*)rawData;
-                for(int y = 0; y < _imageHeight; y++)
+                for(int i = 0; i < image.Length; i++)
                 {
-                    for(int x = 0; x < _imageWidth; x++)
-                    {
-                        image[y, x] = *ushortData / scaleFactor;
-                        ushortData++;
-                    }
+                    image[i] = *ushortData / scaleFactor;
```

Comment:
This division is not required for 50% of the calls to this method.
Would it make sense to have a conditional for that?

Reply:
Done.